# HG changeset patch
# User Piotr Held <pjheld@gmail.com>
# Date 1506645168 25200
#      Thu Sep 28 17:32:48 2017 -0700
# Node ID 3d1b40992075f0188a6b7d48ab87eb090afcb44a
# Parent  46440078d73baed9394481e450b5ca05c997a201
Don't unregister 'meta' package in meta_release (Bug #52111)

* ov-classdef.cc (cdef_package::cdef_package_rep::meta_release):
Done unregister the "meta" package.

diff --git a/libinterp/octave-value/ov-classdef.cc b/libinterp/octave-value/ov-classdef.cc
--- a/libinterp/octave-value/ov-classdef.cc
+++ b/libinterp/octave-value/ov-classdef.cc
@@ -3333,17 +3333,19 @@ cdef_package::cdef_package_rep::meta_rel
   //        could still be referenced by classes or sub-packages?
   //        If the package object is recreated later on, it won't
   //        match the one already referenced by those classes or
   //        sub-packages.
 
   cdef_manager& cdm
     = octave::__get_cdef_manager__ ("cdef_package::cdef_package_rep::meta_release");
 
-  cdm.unregister_package (wrap ());
+  // Don't delete the "meta" package.
+  if (this != cdm.meta ().get_rep ())
+    cdm.unregister_package (wrap ());
 }
 
 //----------------------------------------------------------------------------
 
 cdef_manager::cdef_manager (octave::interpreter& interp)
   : m_interpreter (interp), m_all_classes (), m_all_packages (),
     m_meta_class (), m_meta_property (), m_meta_method (),
     m_meta_package (), m_meta ()
